fhir-codegen/gradle-plugin/src/main/kotlin/com/google/fhir/codegen/ModelTypeSpecGenerator.kt

```diff
@@ -238,25 +241,31 @@ class ModelTypeSpecGenerator(
               enclosingModelClassName.nestedClass(name),
               backboneElements,
               structureDefinition,
-              surrogateTypeSpec,
-              serializerTypeSpec,
+              surrogateTypeSpecGenerator,
+              surrogateFileSpec,
+              serializerFileSpec,
               createBindingToEnumTypeSpecEntry,
             )
             .addSealedInterfaces(
               backboneElementClassName,
               structureDefinition.getElements(backboneElementClassName),
+              serializerFileSpec,
             )
             .build()
         )
 
         // TODO: Handle cases where the BackboneElement does not need the surrogate class and
         //  the custom serializer since it does not have any primitive fields.
-        surrogateTypeSpec.addType(
-          SurrogateTypeSpecGenerator(valueSetMap)
-            .generate(enclosingModelClassName.nestedClass(name.capitalized()), elements)
-        )
-        serializerTypeSpec.addType(
-          SerializerTypeSpecGenerator.generate(enclosingModelClassName.nestedClass(name))
+        val className = enclosingModelClassName.nestedClass(name.capitalized())
```

Comment:
`name` is already capitalized.

also, this className should probably be extracted as a local variable since it's been used before?

Reply:
Resolved.